### PR TITLE
Clean up after unit tests

### LINKED ISSUE #4048: Files used in unit tests remain in `/tmp`
### Describe the bug

After `rake test`, the following files remained in `/tmp`.

* `/tmp/SERVERENGINE_SOCKETMANAGER_*`
* `/tmp/fluentd-lock-*`
* `/tmp/fluent/`

### To Reproduce

1. `rake test`
2. `ls /tmp`

### Expected behavior

Files created by tests should be deleted.

### Your Environment

```markdown
- Fluentd version: 1.15.3
- Operating system: Ubuntu 22.04
- Kernel version: 5.15.0-58-generic
```

### Your Configuration

```apache
-
```

### Your Error Log

```shell
-
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by abetomo**
**Which issue(s) this PR fixes**: 

Fixes #4048

**What this PR does / why we need it**: 

After unit testing, clean up in /tmp.

(If you cannot terminate with SIGTERM and terminate with SIGKILL, some files cannot be cleaned up. https://github.com/fluent/fluentd/pull/4050#issuecomment-1427932016 https://github.com/fluent/fluentd/pull/4050#issuecomment-1430665345)

**Docs Changes**:

None

**Release Note**: 

None

### CODE REVIEW

File: lib/fluent/test/driver/base.rb [hunk 1]
```diff
@@ -173,7 +173,7 @@ def instance_shutdown(log: Logger.new($stdout))
 
           if @socket_manager_server
             @socket_manager_server.close
-            if @socket_manager_server.is_a?(String) && File.exist?(@socket_manager_path)
+            if @socket_manager_path.is_a?(String) && File.exist?(@socket_manager_path)
```
Comment (daipom) [inline lib/fluent/test/driver/base.rb:176]: Wow, thanks! This is a typo!

File: test/plugin/test_in_tcp.rb [hunk 2]
```diff
@@ -226,6 +226,8 @@ def create_tcp_socket(host, port, &block)
 
       assert_equal 1, d.instance.log.logs.count { |l| l =~ /anonymous client/ }
       assert_equal 0, d.events.size
+
+      d.instance_shutdown
```
Comment (daipom) [inline test/plugin/test_in_tcp.rb:230]: Indeed. Need it since the driver ran with `shutdown: false`.

Comment (ashie) [reply test/plugin/test_in_tcp.rb:230]: I think we shouldn't use `shutdown: false` as possible although I'm not sure whether it's needed here or not.
FYI sometimes I found some needless `shutdown: false` in other tests (probably copy&pasted).

Comment (daipom) [reply test/plugin/test_in_tcp.rb:230]: When terminating a plugin on shutdown, the `reset` of the logger is called and it erases the logs.

So this test sets `shutdown: false`.

But somehow the data of `driver.logs` is still alive after terminating...

Comment (daipom) [reply test/plugin/test_in_tcp.rb:230]: So I think using `driver.logs` is the correct way, but I need more time to check how it works.

File: test/plugin/test_output.rb [hunk 3]
```diff
@@ -820,6 +823,7 @@ def waiting(seconds)
       assert try_write_called
 
       i.stop; i.before_shutdown; i.shutdown; i.after_shutdown; i.close; i.terminate
+      FileUtils.rm_rf(tmp_dir)
```
Comment (ashie) [inline test/plugin/test_output.rb:826]: It should be done in `ensure` block or `teardown` to make sure to delete it even if assertions fails.

File: test/plugin/test_in_tcp.rb (hunk 2 above)
Comment (abetomo) [reply test/plugin/test_in_tcp.rb:230]: Is it better to fix it in this PR?
Or should I fix it in another PR?

**Comment by abetomo** [comment]
With the committed changes and the following changes, all files were cleaned.

```diff
diff --git a/test/command/test_fluentd.rb b/test/command/test_fluentd.rb
index 461e19ef..9d62cd40 100644
--- a/test/command/test_fluentd.rb
+++ b/test/command/test_fluentd.rb
@@ -85,12 +85,12 @@ class TestFluentdCommand < ::Test::Unit::TestCase
         yield pid, io
         # p(here: "execute command", pid: pid, worker_pids: @worker_pids)
       ensure
-        Process.kill(:KILL, pid) rescue nil
+        Process.kill(:TERM, pid) rescue nil
         if @supervisor_pid
-          Process.kill(:KILL, @supervisor_pid) rescue nil
+          Process.kill(:TERM, @supervisor_pid) rescue nil
         end
         @worker_pids.each do |cpid|
-          Process.kill(:KILL, cpid) rescue nil
+          Process.kill(:TERM, cpid) rescue nil
         end
         # p(here: "execute command", pid: pid, exist: process_exist?(pid), worker_pids: @worker_pids, exists: @worker_pids.map{|i| process_exist?(i) })
         Timeout.timeout(10){ sleep 0.1 while process_exist?(pid) }
```

When you exit with SIGTERM, the shutdown process runs and the files are deleted.
However, the test was not stable because the process could not terminate in some cases.

File: test/plugin/test_in_tcp.rb (hunk 2 above)
Comment (ashie) [reply test/plugin/test_in_tcp.rb:230]: It's fine as is (`shutdown: false`).
Instead it should be run in ensure block to make sure to stop:
```suggestion
  ensure
      d.instance_shutdown if d && d.instance
```

**Comment by ashie** [comment]
How about trying `SIGTERM` first then `SIGKILL` after timeout?
Although we should make it stable but not needed to do it in this PR.

**Comment by abetomo** [comment]
@ashie  Thanks!
I’ll give it a try.

**Review by ashie — APPROVED** [review]
LGTM

**Comment by ashie** [comment]
Retrying some unstable tests not concerned with this PR...

**Review by daipom — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by ashie** [comment]
Thanks!

**Comment by abetomo** [comment]
Thanks for your review!

File: test/plugin/test_in_tcp.rb (hunk 2 above)
Comment (daipom) [reply test/plugin/test_in_tcp.rb]: Now I understand this.
I made an issue.

* #4136

Comment (daipom) [reply test/plugin/test_in_tcp.rb]: I made a PR

* #4138

**Pull request merged**